Remove PID file at the very end

hebasto opened the pull request:
While reproducing the bug from #18517, I've noticed that the `bitcoind.pid` file has already been removed when the `bitcoind` hangs.

This PR makes `Shutdown()` keep the `bitcoind.pid` file available until the end.

hebasto committed:
Remove PID file at the very end

emilengler commented on src/init.cpp:
```diff
@@ -293,6 +286,14 @@ void Shutdown(NodeContext& node)
     if (node.mempool) node.mempool = nullptr;
     node.scheduler.reset();
     LogPrintf("%s: done\n", __func__);
```
I would put line 288 at the very end.
Done should always be the last message in a log. Otherwise it would be weird if stuff still follows

hebasto replied:
[Done](https://github.com/bitcoin/bitcoin/pull/18526#issuecomment-609823729).

cvengler reviewed (changes requested):
Concept ACK with exception to move the done LogPrintf

hebasto commented:
Updated b2486a3493516745a452b5b87e44289dfb55fed3 -> 7fcdec0f326f7fb547ec9c651871842cf518ea38 ([pr18526.01](https://github.com/hebasto/bitcoin/commits/pr18526.01) -> [pr18526.02](https://github.com/hebasto/bitcoin/commits/pr18526.02), [diff](https://github.com/hebasto/bitcoin/compare/pr18526.01..pr18526.02)):

- addressed @emilengler 's [comment](https://github.com/bitcoin/bitcoin/pull/18526#discussion_r404047165)
> I would put line 288 at the very end.
Done should always be the last message in a log. Otherwise it would be weird if stuff still follows

cvengler reviewed (approved):
utACK 7fcdec0f326f7fb547ec9c651871842cf518ea38

MarcoFalke commented:
ACK 7fcdec0f326f7fb547ec9c651871842cf518ea38

promag reviewed (commented):
Code review ACK 7fcdec0f326f7fb547ec9c651871842cf518ea38.

theStack reviewed (approved):
Code review ACK 7fcdec0f326f7fb547ec9c651871842cf518ea38
